### PR TITLE
Added the documentation of This pointer

### LINKED ISSUE #7111: This pointer in C++
### Description

I will add the documentation of This pointer in C++

### Domain

C/CPP

### Type of Contribution

Documentation

### Code of Conduct

- [X] I follow [Contributing Guidelines](https://github.com/girlscript/winter-of-contributing/blob/main/.github/CONTRIBUTING.md) & [Code of conduct](https://github.com/girlscript/winter-of-contributing/blob/main/.github/CODE_OF_CONDUCT.md) of this project.

### PR DESCRIPTION
**Pull request by Aarushijain-06**
<hr>

##  This Pointer

I have added the documentation of This Pointer in C++
Fixes #7111

<hr>

## Type of change 📝

- [ ] Au#dio (Should be in mp3 format Includes speech clarity, Concise ,Low distortion)
- [ ] Vi#deo (Animations, screen-recordings, presentations and regular explanatory films are all possibilties etc)
- [X] Documentation (Content Creation in the form of codes or tutorials)
- [ ] Other (If you choose other, Please mention changes below) 

<hr>

## Domain of Contribution 📊

- [X] C/CPP

<hr>

## Checklist ✅

- [X] I follow [Contributing Guidelines](https://github.com/girlscript/winter-of-contributing/blob/main/.github/CONTRIBUTING.md) & [Code of conduct](https://github.com/girlscript/winter-of-contributing/blob/main/.github/CODE_OF_CONDUCT.md) of this project.
- [X] I have performed a self-review of my own code or work.
- [X] I have commented my code, particularly in hard-to-understand areas.
- [X] My changes generates no new warnings.
- [X] I'm GWOC'21 contributor

<hr>

### CODE REVIEW

File: C_CPP/Pointer and its types/This pointer.md [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+int main(){
+    B num;
+    num.set_Data(10).get_Data();    // here set_Data() is returning an object, so we have used get_Data() with it only.
+    return 0;
+}
+```
+Output for above program:
+```c++
+10
+```
```
Comment (apeksha235) [inline C_CPP/Pointer and its types/This pointer.md:75]: add a relative link of the file in the readme of the respective folder

**Review by apeksha235 — CHANGES_REQUESTED** [review]
@Aarushijain-06 Also add your file inside ```Pointer and its types/Types of pointers```  and make these changes

**Comment by Aarushijain-06** [comment]
@apeksha235 I have done the changes, please review it now.😊

**Comment by apeksha235** [comment]
@Aarushijain-06 please add the file in the folder I mentioned, don't make a new folder for it.

**Comment by Aarushijain-06** [comment]
Done with all the changes

File: C_CPP/Pointer and its types/Types of Pointers/This_pointer.md [hunk 2]
```diff
@@ -0,0 +1,75 @@
+# This Pointer in C++
+
+### Definition: This is a keyword which is a pointer that points to the object which invokes the member function.
```
Comment (apeksha235) [inline C_CPP/Pointer and its types/Types of Pointers/This_pointer.md:3]: @Aarushijain-06 one last thing, just put the word ```this``` in bold or highlight it to give it more emphasis

**Review by apeksha235 — CHANGES_REQUESTED** [review]

File: C_CPP/Pointer and its types/Types of Pointers/This_pointer.md (hunk 2 above)
Comment (Aarushijain-06) [reply C_CPP/Pointer and its types/Types of Pointers/This_pointer.md:3]: Actually I am out for some days, and did not have my laptop😪 soo can i perform the change after few days but at that time the issue will get closed.

Comment (apeksha235) [reply C_CPP/Pointer and its types/Types of Pointers/This_pointer.md:3]: @Aarushijain-06 how many days?

Comment (Aarushijain-06) [reply C_CPP/Pointer and its types/Types of Pointers/This_pointer.md:3]: I will return on this 18.

Comment (apeksha235) [reply C_CPP/Pointer and its types/Types of Pointers/This_pointer.md:3]: @Aarushijain-06 tag me as soon as you make the changes, I will review it.

Comment (Aarushijain-06) [reply C_CPP/Pointer and its types/Types of Pointers/This_pointer.md:3]: Okay ma'am!

**Comment by Aarushijain-06** [comment]
@apeksha235  ma'am, I have made the change, please review it.

**Review by apeksha235 — APPROVED** [review]
@Aarushijain-06 good work! 
@Saviour1001 can be merged!

**Pull request merged**